### PR TITLE
feat(aws/eks): add new and move jobs nodes to single-az

### PR DESCRIPTION
**Pull request by maxgio92**
> Note: this will re-create all the node groups, besides adding a new one deployed in single-AZ.

/cc @zuc @FedeDP @cappellinsamuele

### CODE REVIEW

File: config/clusters/eks_variables.tf [hunk 1]
```diff
@@ -14,7 +16,7 @@ variable "eks_default_worker_group_instance_type" {
 }
 
 variable "eks_default_worker_group_asg_min_capacity" {
-  default     = 3
+  default     = 1
```
Comment (maxgio92) [inline config/clusters/eks_variables.tf:19]: The minimum of one node could be enough to run Prow microservices-only.
It's anyway auto-scaled by the cluster-autoscaler deployed on Kubernetes.

**Review by FedeDP — COMMENTED** [review]
/lgtm

**Comment by maxgio92** [comment]
Note: right now jobs miss the toleration for the singleaz nodes taint

**Comment by maxgio92** [comment]
Integrated by #1136

**Comment by maxgio92** [comment]
Ready to be reviewed @FedeDP

File: config/clusters/eks.tf [hunk 2]
```diff
@@ -40,29 +40,62 @@ module "eks" {
...
+
+    jobs_arm = {
+      name             = var.eks_jobs_arm_worker_group_name
+      desired_capacity = var.eks_jobs_arm_worker_group_asg_desired_capacity
+      max_capacity     = var.eks_jobs_arm_worker_group_asg_max_capacity
+      min_capacity     = var.eks_jobs_arm_worker_group_asg_min_capacity
+      instance_types   = [var.eks_jobs_arm_worker_group_instance_type]
+
+      ami_type = "AL2_ARM_64"
+      taints   = [local.arm_nodegroup_taint]
```
Comment (FedeDP) [inline config/clusters/eks.tf:86]: Shouldn't this have also the `local.single_az_nodegroup_taint`?

File: config/clusters/eks_variables.tf [hunk 3]
```diff
@@ -30,20 +32,64 @@ variable "eks_default_worker_group_asg_max_capacity" {
...
+  description = "The Autoscaling Group size of the jobs EKS worker group"
+}
+
+variable "eks_jobs_worker_group_asg_desired_capacity" {
+  default     = 4
+  description = "The Autoscaling Group size of the jobs EKS worker group"
+}
+
+variable "eks_jobs_worker_group_asg_max_capacity" {
```
Comment (FedeDP) [inline config/clusters/eks_variables.tf:60]: Shouldn't we upload namings in prow.tfvars too? Ie: https://github.com/falcosecurity/test-infra/blob/master/config/clusters/prow.tfvars#L10C50-L12

File: config/clusters/eks.tf (hunk 2 above)
Comment (maxgio92) [reply config/clusters/eks.tf:86]: Nope @FedeDP. It's deployed in multi-AZ here

File: config/clusters/eks_variables.tf (hunk 3 above)
Comment (maxgio92) [reply config/clusters/eks_variables.tf:60]: Good catch @FedeDP, thank you.

File: config/clusters/eks.tf (hunk 2 above)
Comment (FedeDP) [reply config/clusters/eks.tf:86]: Mmh so build drivers arm64 jobs will still risk to be evicted?

File: config/clusters/eks_variables.tf (hunk 3 above)
Comment (maxgio92) [reply config/clusters/eks_variables.tf:60]: Whoops, sorry, I misunderstood. We don't need it, we have default values right now

File: config/clusters/eks.tf (hunk 2 above)
Comment (maxgio92) [reply config/clusters/eks.tf:86]: I didn't see the amount of eviction for arm builds right now. But we can do it also for them. WDYT?

Comment (cappellinsamuele) [reply config/clusters/eks.tf:86]: I can confirm that we experienced some evictions also from arm nodes in one of the last runs, so maybe it's a good idea to add the taint

Comment (maxgio92) [reply config/clusters/eks.tf]: Applied @FedeDP @cappellinsamuele :)

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**